services/tables/src/test/java/com/linkedin/openhouse/tables/e2e/h2/RepositoryTest.java

```diff
@@ -501,14 +501,30 @@ void testSchemaEvolutionBasic() {
 
     Assertions.assertThrows(
         InvalidSchemaEvolutionException.class,
-        () ->
-            validator.validateWriteSchema(
-                oldSchema, newSchema, update.apply(), createDto.getTableUri()));
+        () -> validator.validateWriteSchema(oldSchema, newSchema, createDto.getTableUri()));
     TableDtoPrimaryKey primaryKey = getPrimaryKey(TABLE_DTO);
     openHouseInternalRepository.deleteById(primaryKey);
     Assertions.assertFalse(openHouseInternalRepository.existsById(primaryKey));
   }
 
+  @Test
+  void testSchemaEvolutionWithMismatchedFieldId() {
```

Comment:
can you also add another test which has 2 subsequent evolutions. That will help validate that the evolved schema from 1st evolution is not corrupted.